### PR TITLE
Add custom SB properties & gzip compression support

### PR DESCRIPTION
**Pull request by harreman**
Enable configuration of custom Service Bus application properties per destination, with validation to prevent reserved key overrides. Add optional gzip compression for message bodies, automatically setting Content-Encoding. Update orchestrator and publisher logic to merge properties and handle compression. Include new unit and integration tests for validation, property merging, and compression utilities.

### CODE REVIEW

File: src/FileHorizon.Application/Infrastructure/Messaging/ServiceBus/AzureServiceBusFileContentPublisher.cs [hunk 1]
```diff
@@ -107,33 +107,55 @@ private static Result ValidateRequest(FilePublishRequest request)
...
+                messageContent = compressed;
+                isCompressed = true;
+                _logger.LogDebug("Compressed {FileName} content from {OriginalSize} to {CompressedSize} bytes",
+                    request.FileName, request.Content.Length, compressed.Length);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Failed to compress content for {FileName}; sending uncompressed", request.FileName);
+                // Continue with uncompressed content
+            }
```
[the remark this answers is not in the document]
Comment (harreman) [reply src/FileHorizon.Application/Infrastructure/Messaging/ServiceBus/AzureServiceBusFileContentPublisher.cs:149]: Fixed in 958f48b. Added `catch (OperationCanceledException) { throw; }` before the generic `catch (Exception)` block so cancellation propagates correctly instead of being swallowed and mislogged as a compression failure.

File: src/FileHorizon.Application/Configuration/DestinationsOptionsValidator.cs [hunk 2]
```diff
@@ -39,6 +47,39 @@ public ValidateOptionsResult Validate(string? name, DestinationsOptions options)
...
+            else if (!seenNames.Add(d.Name)) errors.Add($"{prefix}: Name '{d.Name}' is duplicated.");
+            if (string.IsNullOrWhiteSpace(d.EntityName)) errors.Add($"{prefix}: EntityName must be specified.");
+            
+            // Validate authentication configuration
+            var hasConnectionString = !string.IsNullOrWhiteSpace(d.ServiceBusTechnical.ConnectionString);
+            var hasNamespace = !string.IsNullOrWhiteSpace(d.ServiceBusTechnical.FullyQualifiedNamespace);
+            if (!hasConnectionString && !hasNamespace)
+            {
+                errors.Add($"{prefix}: Either ServiceBusTechnical.ConnectionString or ServiceBusTechnical.FullyQualifiedNamespace must be specified.");
+            }
```
[the remark this answers is not in the document]
Comment (harreman) [reply src/FileHorizon.Application/Configuration/DestinationsOptionsValidator.cs]: Fixed in 958f48b. Added a null guard for `ServiceBusTechnical` — if null, the validator now returns a clear error message instead of throwing NullReferenceException. Also added a dedicated unit test (`Validate_ServiceBusDestination_NullServiceBusTechnical_Fails`) covering this path.

File: src/FileHorizon.Application/Infrastructure/Messaging/ServiceBus/AzureServiceBusFileContentPublisher.cs [hunk 3]
```diff
@@ -107,33 +107,55 @@ private static Result ValidateRequest(FilePublishRequest request)
...
+        }
```
[the remark this answers is not in the document]
Comment (harreman) [reply src/FileHorizon.Application/Infrastructure/Messaging/ServiceBus/AzureServiceBusFileContentPublisher.cs:150]: Acknowledged. The `CompressionHelper` class itself has 9 active unit tests (round-trip, cancellation, edge cases). The publisher integration test remains skipped because `CreateMessage` is private and `SendMessageAsync` requires a live Service Bus connection. Adding a non-skipped test would require extracting message creation behind a testable seam — keeping that as a follow-up refactoring item rather than expanding scope in this PR.

**Review by LittleAndi — APPROVED** [review]
